packages/chart/src/Contour.ts

```diff
@@ -0,0 +1,109 @@
+import { contourDensity as d3ContourDensity } from "d3-contour";
+import { geoPath } from "d3-geo";
+import { XYAxis } from "./XYAxis";
+
+export class Contour extends XYAxis {
+
+    constructor() {
+        super();
+        this
+            .xAxisGuideLines_default(false)
+            .yAxisGuideLines_default(false)
+            ;
+    }
+
+    xPos(host: XYAxis, d) {
+        return host.orientation() === "horizontal" ? host.dataPos(d.label) : host.valuePos(d.value);
+    }
+
+    yPos(host: XYAxis, d) {
+        return host.orientation() === "horizontal" ? host.valuePos(d.value) : host.dataPos(d.label);
+    }
+
+    layerEnter(host: XYAxis, element, duration: number = 250) {
+        super.layerEnter(host, element, duration);
+    }
+
+    layerUpdate(host: XYAxis, element, duration: number = 250) {
+        super.layerUpdate(host, element, duration);
+
+        const data = this.flattenData();
+
+        const contourData = d3ContourDensity()
+            .x(d => this.xPos(host, d))
+            .y(d => this.yPos(host, d))
+            .size([this.width(), this.height()])
+            .bandwidth(this.contourBandwidth())
+            (data)
+            ;
+        const lines = element
+            .selectAll("path")
+            .data(contourData)
+            ;
+        lines.enter().append("path")
+            .attr("d", geoPath())
+            .attr("fill", this.contourFillColor())
+            .attr("stroke", this.contourStrokeColor())
+            .attr("stroke-linejoin", "round")
+            .attr("stroke-width", this.contourStrokeWidth())
+            ;
+        lines
+            .attr("d", geoPath())
+            .attr("fill", this.contourFillColor())
+            .attr("stroke", this.contourStrokeColor())
+            .attr("stroke-linejoin", "round")
+            .attr("stroke-width", this.contourStrokeWidth())
+            ;
```

Comment:
The above (enter and update) can be shortened to the following (merge is new in d3v4):
```
lines.enter().append("path")
            .attr("stroke-linejoin", "round")
        .merge(lines)  //  Merges the just added "enter" items with the "update" items
            .attr("d", geoPath())
            .attr("fill", this.contourFillColor())
            .attr("stroke", this.contourStrokeColor())
            .attr("stroke-width", this.contourStrokeWidth())
            ;
```